### ISSUE TITLE
Results summary

### ISSUE THREAD

**Issue by phene** [aws-cloudformation/cfn-python-lint#882]
*cfn-lint version: 0.20.0*

It would be great it cfn-lint supported a summary of results about the current run, including:

* How many files were checked
* How many rules failed.

Require flag to enable it.

**Comment by luistapia6-11** [comment]
+1 It would great to get more information at the end of a test. How many rules were checked, how many rules passed/failed. Similar to running an Ansible playbook.

**Comment by fatbasstard** [comment]
Actually like this one. I've been hacking some counters in the past to find the amount of files processed.... Making it a feature is 🎉

**Comment by jakrob** [comment]
+1 It would great to get summary table showing stats for each error/warning types

**Comment by PatMyron** [comment]
> +1 It would great to get summary table showing stats for each error/warning types

@jakrob I've been piping the output into something like this to get the count for rules flagged:

```shell
$ cfn-lint template.yaml | grep -Eo '^\w\d+' | sort | uniq -c | sort -nr
  20 W3011
  16 E3008
  16 E3002
   8 E2531
   8 E2001
   4 W2511
   4 W2501
   1 E0000
```
